deno_dist/utils/url.ts

```diff
@@ -70,10 +70,25 @@ export const getPattern = (label: string): Pattern | null => {
 }
 
 export const getPath = (request: Request): string => {
-  // Optimized: indexOf() + slice() is faster than RegExp
   const url = request.url
-  const queryIndex = url.indexOf('?', 8)
-  return url.slice(url.indexOf('/', 8), queryIndex === -1 ? undefined : queryIndex)
+  const start = url.indexOf('/', 8)
+  let i = start
+  for (; i < url.length; i++) {
+    const charCode = url.charCodeAt(i)
+    if (charCode === 37) {
+      // '%'
+      // If the path contains percent encoding, use `indexOf()` to find '?' and return the result immediately.
+      // Although this is a performance disadvantage, it is acceptable since we prefer cases that do not include percent encoding.
+      const queryIndex = url.indexOf('?', i)
+      return decodeURI(
+        url.slice(start, queryIndex === -1 ? undefined : queryIndex).replace(/%25/g, '%2525')
```

Comment:
From my benchmarks it seems like `str.includes('%25') ? str.replaceAll('%25', '%2525') : str` is faster than calling `.replace` with regex. Can you verify?

Reply:
Thank you. You are right!

It's faster to check in `.includes` first.
`s/replace/replaceAll/` didn't change in my benchmark results, so I'll leave `replace(//g)` as it is, as it is generally used elsewhere in Hono.

Applied in 0eeaa7ec